backend/open_webui/retrieval/utils.py

```
@@ -681,6 +700,60 @@ def generate_openai_batch_embeddings(
+                json=json_data,
+            )
+            if r.status_code == 429:
+                retry = float(r.headers.get("Retry-After", "1"))
+                time.sleep(retry)
+                continue
+            r.raise_for_status()
+            data = r.json()
+            if "data" in data:
+                return [elem["embedding"] for elem in data["data"]]
+            else:
+                raise Exception("Something went wrong :/")
```

Ruff (0.11.9)

Abstract `raise` to an inner function

(TRY301)